### PR TITLE
home: respect user record nosuid, nodev and noexec flags

### LINKED ISSUE #15892: systemd-homed doesn't respect mount flags (nosuid, nodev, exec)
**systemd version the issue has been seen with**

`systemd 245 (245.5-2-arch)
+PAM +AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid`

**Used distribution**

Arch Linux

**Expected behaviour you didn't see**

```bash
$ homectl update --nosuid=false <username>
$ mount | grep <username> # shouldn't show nosuid
```

`homectl inspect` after the update shows that the option has been correctly set.

**Unexpected behaviour you saw**

`mount | grep <username>` shows nosuid even after updating my user to not use nosuid and deactivating and activating my home.

**Steps to reproduce the problem**

Create a user using `systemd-homed` with the LUKS storage backend and `--nosuid=false`. Alternatively, update it with `homectl update --nosuid=false <username>`.

Deactivate and reactivate the home.

Check `mount | grep <username>`

Extra info: https://github.com/systemd/systemd/blob/0aeb407cb4561704e361936ac6d5afc92cd50fd8/src/home/homework-mount.c#L41 seems like the options are just ignored.

### PR DESCRIPTION
**Pull request by tudurom**
This fixes #15892.

### CODE REVIEW

File: src/home/homework-cifs.c [hunk 1]
```diff
@@ -28,7 +28,7 @@ int home_prepare_cifs(
                 char **pw;
                 int r;
 
-                r = home_unshare_and_mount(NULL, NULL, false);
+                r = home_unshare_and_mount(NULL, NULL, false, h->nosuid, h->nodev, h->noexec);
```
Comment (poettering) [inline src/home/homework-cifs.c]: hmm, please pass these as flags field, i.e. pass in user_record_mount_flags().

the simple directory backend, implemented in `src/home/homework-directory.c` actually did this right, it's the only backend currently using user_record_mount_flags() correctly for this. We should really do this the same way for the other backends too.

**Comment by tudurom** [comment]
I didn't know about that function, it sure is a cleaner way of doing this.

I will rework this tomorrow, it's my first contribution. Thanks for the help!

**Comment by tudurom** [comment]
Reworked the PR to use `user_record_mount_flags()` for the affected backends (LUKS, CIFS).

**Comment by poettering** [comment]
Looks excellent! Thank you!

**Pull request merged**